### PR TITLE
Always return bytes from enc()

### LINKED ISSUE #26: transition to FFI broke set(path, None)
As far as I can tell handling `None` values require special care. Right now an attempt to set a value to `None` produces following error:

```
t._aug.set("/files/tmp/hosts/#comment[4]", None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-d057463c301f> in <module>()
----> 1 t._aug.set("/files/tmp/hosts/#comment[4]", None)


/usr/lib/python3.6/site-packages/augeas/__init__.py in set(self, path, value)
    166 
    167         # Call the function
--> 168         ret = lib.aug_set(self.__handle, enc(path), enc(value))
    169         if ret != 0:
    170             raise ValueError("Unable to set value to path!")


TypeError: initializer for ctype 'char *' must be a bytes or list or tuple, not str
```

This is very unfortunate because setting values to None is used to implement `touch` command similar to `augtool`.

### PR DESCRIPTION
**Pull request by yan12125**
Fix #26

### CODE REVIEW

**Comment by pspacek** [comment]
Tests can be re-run when Python 3.2 support is removed.

**Comment by felixonmars** [comment]
Confirmed that this patches fixes certbot-apache's test suite for me.

**Comment by pspacek** [comment]
#29 was merged so this PR will most likely pass CI tests when it is rebased on top of current master (to remove the Python 3.2 from .travis-cii.yml).

**Comment by yan12125** [comment]
Thanks @pspacek! I rebased this PR and Travis is satisfying :)

File: augeas/__init__.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def enc(st):
     if st:
         return st.encode(AUGENC)
     else:
-        return ''
+        return b''
 
 
 def dec(st):
```
Comment (pspacek) [inline augeas/__init__.py]: Hmm, I guess that `dec()` should have the same logic. Please see `return ''` below.

**Comment by thedrow** [comment]
Should be fixed in 1.0.3 which is being released now :)

**Comment by pspacek** [comment]
Have you checked the `dec()` as well? The same fix might be missing in there.

**Comment by thedrow** [comment]
I fixed dec() as well.

**Comment by pspacek** [comment]
Great, thanks!

**Comment by yan12125** [comment]
Thanks!

**Pull request merged**